audit/log_test.go

```diff
@@ -0,0 +1,205 @@
+		So(eventData, ShouldContainKey, "error")
+		So(eventData["error"], ShouldHaveSameTypeAs, errors.New(""))
+		So(eventData["error"].(error).Error(), ShouldEqual, "test error")
+		So(eventData, ShouldNotContainKey, "caller")
+		So(eventData, ShouldNotContainKey, "user")
+
+		eventData = log.Data{}
+	})
+
+	Convey("context with user", t, func() {
+		contextWithUser := context.WithValue(ctx, common.UserIdentityKey, "user email")
+		LogError(contextWithUser, errors.New("test error"), nil)
+		So(eventName, ShouldEqual, "error")
+		So(eventCorrelationKey, ShouldEqual, "")
+		So(eventData, ShouldContainKey, "message")
+		So(eventData["message"], ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "error")
+		So(eventData["error"], ShouldHaveSameTypeAs, errors.New(""))
+		So(eventData["error"].(error).Error(), ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "user")
+		So(eventData["user"], ShouldHaveSameTypeAs, "")
+		So(eventData["user"], ShouldEqual, "user email")
+		So(eventData, ShouldNotContainKey, "caller")
+
+		eventData = log.Data{}
+	})
+
+	Convey("context with caller id", t, func() {
+		contextWithCaller := context.WithValue(ctx, common.CallerIdentityKey, "api service")
+		LogError(contextWithCaller, errors.New("test error"), nil)
+		So(eventName, ShouldEqual, "error")
+		So(eventCorrelationKey, ShouldEqual, "")
+		So(eventData, ShouldContainKey, "message")
+		So(eventData["message"], ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "error")
+		So(eventData["error"], ShouldHaveSameTypeAs, errors.New(""))
+		So(eventData["error"].(error).Error(), ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "caller")
+		So(eventData["caller"], ShouldHaveSameTypeAs, "")
+		So(eventData["caller"], ShouldEqual, "api service")
+		So(eventData, ShouldNotContainKey, "user")
+
+		eventData = log.Data{}
+	})
+
+	Convey("context with request, user and caller ids", t, func() {
+		ctx := getContextWithCallerUserAndRequestIDcontext()
+		LogError(ctx, errors.New("test error"), nil)
+		So(eventName, ShouldEqual, "error")
+		So(eventCorrelationKey, ShouldEqual, "request id")
+		So(eventData, ShouldContainKey, "message")
+		So(eventData["message"], ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "error")
+		So(eventData["error"], ShouldHaveSameTypeAs, errors.New(""))
+		So(eventData["error"].(error).Error(), ShouldEqual, "test error")
+		So(eventData, ShouldContainKey, "caller")
+		So(eventData["caller"], ShouldHaveSameTypeAs, "")
+		So(eventData["caller"], ShouldEqual, "api service")
+		So(eventData, ShouldContainKey, "user")
+		So(eventData["user"], ShouldHaveSameTypeAs, "")
+		So(eventData["user"], ShouldEqual, "user email")
+
+		eventData = log.Data{}
+	})
+}
+
+func TestAuditInfoLog(t *testing.T) {
+
+	var eventName, eventCorrelationKey string
+	var eventData log.Data
+	log.Event = func(name string, correlationKey string, data log.Data) {
+		eventName = name
```

Comment:
the tests look long & opaque to me,  I'd like to see use of a struct:
e.g.
```
// top of package
type eventResults struct { name string, cKey string, data log.Data }
...
// above this func
var eventResult eventResults
...
// in this func
eventResult = eventResults{name: name, cKey: correlationKey, data: data}
...
// in tests...
So(eventResult, ShouldResemble, eventResults{
          name: "info", 
          cKey: "",
          data: log.Data{message:"info message"},
})
...
// at the end of each test (also fixes the problem that you're not resetting e.g. eventName)
eventResult = eventResults{}
```
?